File: category/greedy/PGS_42883_큰 수 만들기.js
function solution(number, k) {
  let answer = "";
  const stack = [];

  for (let i = 0; i < number.length; i++) {
    const num = number[i];

    while (stack.length > 0 && stack[stack.length - 1] < num && k > 0) {
      stack.pop();
      k--;
    }
    stack.push(num);
  }

  if (k > 0) {
    stack.splice(stack.length - k, k); //배열에서 원하는 인덱스부터 k개를 제거하는 메서드
  }

  answer = stack.join("");
  return answer;
}

File: category/greedy/PGS_42883_큰 수 만들기.py

def solution(number, k):
    answer = ''
    max_index = -1
    k = len(number)-k
    while k > 0:
        max = -1
        for i in range (max_index+1,len(number) - k + 1):
            if max < int(number[i]):
                max = int(number[i])
                max_index = i
                if max == '9':
                 break
        k -= 1
        answer += number[max_index]

        print('max_index:',max_index)

        print('k',k)
        print('answer:',answer)

    return answer

print(solution('1231234',3))
#테스트 케이스 10번 시간 초과 오류. 시간복잡도 O(N^2)라서...